e2etest/l2tests/node_service.go

```diff
@@ -15,22 +15,22 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-func nodeForService(svc *corev1.Service, nodes []corev1.Node) (string, error) {
+func nodeForService(svc *corev1.Service, nodes []corev1.Node) (*corev1.Node, error) {
```

Comment:
can you explain why do we need this refactor (possibly in the commit message, and maybe in a separate commit if it's not completely needed for this new test)?